GH-19991: fix styling for channel-archived-warning

kyeongsoosoo opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

Fixed spacing and Added a divider.
`css` for `.channel-archived-warning` was included  in `.sidebar--right`.
I moved it to `.ThreadViewer`, so styling can be applied to global thread view.



#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

#19991

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.

For an easier comparison of UI changes a table (template below) can be used.

|  before  |  after  |
|----|----|
| <insert before screenshot here> |  |

-->

|  before  |  after  |
|----|----|
| <img width="400" alt="스크린샷 2022-04-23 오전 4 27 11" src="https://user-images.githubusercontent.com/59413880/164781348-8662b001-653c-4234-bf22-69fe0d9d1f3e.png">|<img width="400" alt="스크린샷 2022-04-23 오전 3 55 53" src="https://user-images.githubusercontent.com/59413880/164780950-30aa9d2e-59d6-4895-b7f2-50989f79c269.png">|


#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates)
* API additions—new endpoint, new response fields, or newly accepted request parameters
* Database changes (any)
* Websocket additions or changes
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating)
* New features and improvements, including behavioural changes, UI changes and CLI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE
```
-->
```release-note
None
```

mattermod commented:
Hello @kyeongsoosoo,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

hmhealey reviewed (approved):
Thanks for the PR, @kyeongsoosoo! This looks good to me

Here's how it looks now.
![Screen Shot 2022-04-25 at 12 07 03 PM](https://user-images.githubusercontent.com/3277310/165128947-d8255187-a855-492b-946f-2a66927b7ad6.png)

esethna commented:
Thanks @kyeongsoosoo!

Looks like it's missing the icon from the design? https://www.figma.com/file/CI11dCaVGPyZjFo26qfZBF/MM-40445-Global-Threads-Archived-Channel-Threads?node-id=0%3A1

hmhealey commented:
Good catch. We definitely missed that.

@kyeongsoosoo To get that exact icon, you should be able to use a `CompassIcon` component and then provide it the name `archive-icon`

kyeongsoosoo commented:
Thanks @esethna @hmhealey , I missed the icon.
Now it looks
<img width="500" alt="스크린샷 2022-04-29 오후 1 08 13" src="https://user-images.githubusercontent.com/59413880/165884003-cdc53624-e9d2-4b24-b81b-fdda90b3ce60.png">

esethna commented:
Thanks @kyeongsoosoo! Can you help make sure the alignment and spacing is pixel perfect as per the design? https://www.figma.com/file/CI11dCaVGPyZjFo26qfZBF/MM-40445-Global-Threads-Archived-Channel-Threads?node-id=0%3A1

![image](https://user-images.githubusercontent.com/14333569/165978448-1ddf0cc4-8794-4836-b019-7fadb19bcfe0.png)

kyeongsoosoo commented:
@esethna Thanks for letting me know the missing part. Now it's same with the figma design.

matthewbirtch commented:
In the context of a thread, the message should be "You are viewing a thread from an archived channel. New messages cannot be posted." 

This should help make it clearer

Other than that, I think everything is good

kyeongsoosoo commented:
@matthewbirtch Done!
Here's the change
<img width="1059" alt="스크린샷 2022-04-30 오전 5 24 30" src="https://user-images.githubusercontent.com/59413880/166064325-17928548-8bbd-4e70-8054-ea62f883d128.png">

kyeongsoosoo commented:
Should I add translations to pass the i18n test?

hmhealey commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/190376

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

jgilliam17 commented:
/update-branch

kyeongsoosoo commented:
@jgilliam17 
I fixed the `MM-T1680` e2e test.
I changed the className `channel-archived-warning` to `channel-archived-warning__container` and `channel-archived-warning__content`, so I applied this change to the test!

jgilliam17 commented:
@hmhealey Can you please approve the workflow? Thanks

jgilliam17 commented:
Thanks for fixing the test - [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/195065-0f81368-onprem-ent-webapp-pr-10222-mm-ee-test:0f81368/mochawesome.html) looks good, no PR related failures.

amyblais commented:
/update-branch

mattermod commented:
Will try to auto merge this PR once all tests and checks are passing. This might take up to an hour.

mattermod commented:
Trying to auto merge this PR.

mattermod commented:
Pull Request successfully merged
SHA: 078576d92a1ebda2791bb6533ece9368446bee82

amyblais commented:
/cherry-pick cloud
